Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTestEnumerationSuiteFactory.cpp

```diff
@@ -84,4 +87,74 @@ namespace TestImpact
             return testSuites;
         }
     } // namespace GTest
+
+    namespace Python
+    {
+        AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>> TestEnumerationSuitesFactory(
+            const AZStd::string& testEnumerationData)
+        {
+            const char newLineDelim = '\n';
+            size_t startLine;
+            size_t endLine = 0;
+            int count = 0;
+
+            AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>> pairs;
+            AZStd::map<AZStd::string, AZStd::vector<TestEnumerationSuite>> testSuiteMap;
+            AZStd::string previousTestFixture;
+            // Steps through our string line by line. Assumes lines are split by newline characters.
+            while ((startLine = testEnumerationData.find_first_not_of(newLineDelim, endLine)) != AZStd::string::npos)
+            {
+                endLine = testEnumerationData.find(newLineDelim, startLine);
+                AZStd::string curLine = testEnumerationData.substr(startLine, endLine - startLine);
+
+                // This regex matches pytest test names, in the form ScriptPath::TestFixture::TestName. It will accept both parameterized
+                // and unparamaterized tests (with or without [] at the end).
+                AZStd::basic_regex testNamePattern = AZStd::basic_regex("(.*py)::([A-Za-z_/\\-0-9]*)::(.*)");
```

Comment:
suggestion: We may want to hoist this outside the loop (not sure how expensive constructing a regex can be) - actually better might be making this `constexpr` if possibe (or hoisting and making it `const`)

Reply:
I don't think moving it out of the loop will make any difference (it's on the stack and initialized only once regardless of whether it's in the loop or outside) and arguably it's better scoped being in the loop but deffo 👍 on the `constexpr` (or `const` if `constexpr` is not possible).

Reply:
I've made it a const inside the loop - basic_regex doesn't support constexpr unfortunately.